pkg/api/persistentvolumeclaim/util.go

```diff
@@ -52,3 +52,20 @@ func volumeSnapshotDataSourceInUse(oldPVCSpec *core.PersistentVolumeClaimSpec) b
 	}
 	return false
 }
+
+func dataSourceIsEnabled(pvcSpec *core.PersistentVolumeClaimSpec) bool {
+	if pvcSpec.DataSource != nil {
```

Comment:
Do we really need this level of checking?  I mean, is it possible that external controllers could consume this with other types that we don't know about here?

Reply:
IMO no, in fact I thought it seemed redundant and had initially just removed the restrictions on the datasource field with a new feature gate and called it done.  Yes, my thought was this was designed for external controllers and I'd love to be able to use it as such.  

I'm probably not helping myself out here.